ecdsa/src/ecdsa.rs

```diff
@@ -262,27 +283,56 @@ mod tests {
                     let offset = 0;
                     let ctx = &mut RegionCtx::new(region, offset);
 
+                    // r and s should be less than Remiander
+                    let r = if self.valid_input {
+                        let r = format!("{:?}", self.r);
+                        big_uint::from_str_radix(&r[2..], 16).unwrap()
+                    } else {
+                        scalar_chip.rns().max_remainder.clone() + big_uint::from(20u32)
+                    };
+                    let s = if self.valid_input {
+                        let s = format!("{:?}", self.s);
+                        big_uint::from_str_radix(&s[2..], 16).unwrap()
+                    } else {
+                        scalar_chip.rns().max_remainder.clone() + big_uint::from(20u32)
+                    };
+                    let is_r_s_within_ranage = scalar_chip.assign_constant(ctx, ((r <= scalar_chip.rns().max_remainder && s <= scalar_chip.rns().max_remainder) as u64).into())?;
+                    let is_valid = scalar_chip.is_not_zero_without_reduce(ctx, &is_r_s_within_ranage)?;
```

Comment:
I don't understand this part. `is_r_s_within_ranage` is not calculated inside the circuit so can't directly be used, but maybe could be useful for testing, but I see is being used as the initial value below so not sure what's going on here?

When removing this check I don't think you still need to convert to `big_uint` right?